src/managers/GuildBanManager.js

```diff
@@ -0,0 +1,177 @@
+   * @returns {?GuildBan}
+   */
+  resolve(ban) {
+    return super.resolve(ban) ?? super.resolve(this.client.users.resolveID(ban));
+  }
+
+  /**
+   * Options used to fetch a single ban from a guild.
+   * @typedef {Object} FetchBanOptions
+   * @property {UserResolvable} user The ban to fetch
+   * @property {boolean} [cache=true] Whether or not to cache the fetched ban
+   * @property {boolean} [force=false] Whether to skip the cache check and request the API
+   */
+
+  /**
+   * Options used to fetch all bans from a guild.
+   * @typedef {Object} FetchBansOptions
+   * @property {boolean} cache Whether or not to cache the fetched bans
+   */
+
+  /**
+   * Fetches ban(s) from Discord.
+   * @param {UserResolvable|FetchBanOptions|FetchBansOptions} [options] Options for fetching guild ban(s)
+   * @returns {Promise<GuildBan>|Promise<Collection<Snowflake, GuildBan>>}
+   * @example
+   * // Fetch all bans from a guild
+   * guild.bans.fetch()
+   *   .then(console.log)
+   *   .catch(console.error);
+   * @example
+   * // Fetch all bans from a guild without caching
+   * guild.bans.fetch({ cache: false })
+   *   .then(console.log)
+   *   .catch(console.error);
+   * @example
+   * // Fetch a single ban
+   * guild.bans.fetch('351871113346809860')
+   *   .then(console.log)
+   *   .catch(console.error);
+   * @example
+   * // Fetch a single ban without checking cache
+   * guild.bans.fetch({ user, force: true })
+   *   .then(console.log)
+   *   .catch(console.error)
+   * @example
+   * // Fetch a single ban without caching
+   * guild.bans.fetch({ user, cache: false })
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  fetch(options) {
+    if (!options) return this._fetchMany();
+    const user = this.client.users.resolveID(options);
+    if (user) return this._fetchSingle({ user, cache: true });
+    if (options.user) {
+      options.user = this.client.users.resolveID(options.user);
+    }
+    if (!options.user) {
+      if ('cache' in options) return this._fetchMany(options.cache);
+      return Promise.reject(new Error('FETCH_BAN_RESOLVE_ID'));
+    }
+    return this._fetchSingle(options);
+  }
+
+  async _fetchSingle({ user, cache, force = false }) {
+    if (!force) {
+      const existing = this.cache.get(user);
+      if (existing && !existing.partial) return existing;
+    }
+
+    const data = await this.client.api.guilds(this.guild.id).bans(user).get();
+    return this.add(data, cache);
+  }
+
+  async _fetchMany(cache) {
+    const data = await this.client.api.guilds(this.guild.id).bans.get();
+    return data.reduce((col, ban) => col.set(ban.user.id, this.add(ban, cache)), new Collection());
+  }
+
+  /**
+   * Bans a user from the guild.
+   * @param {UserResolvable} user The user to ban
+   * @param {Object} [options] Options for the ban
+   * @param {number} [options.days=0] Number of days of messages to delete, must be between 0 and 7
```

Comment:
```suggestion
   * @param {number} [options.days=0] Number of days of messages to delete, must be between 0 and 7, inclusive
```
#5327 will add this change to only `GuildMember#ban` & `GuildMemberManager#ban`